content/en/platform/corda/4.8/open-source/generating-a-node-cordform.md

```diff
@@ -0,0 +1,303 @@
+
+```kotlin
+sshd {
+  port = 2222
+}
+```
+
+You can extend the `deployNodes` task with more `node {}` blocks to generate as many nodes as necessary for your application.
+
+{{< warning >}}
+When adding nodes, make sure that there are no port clashes!
+{{< /warning >}}
+
+To extend node configuration beyond the properties defined in the `deployNodes` task, use the `configFile` property with the file path (relative or absolute) set to an additional configuration file. This file should follow the standard [Node configuration](corda-configuration-file.md) format of `node.conf`. The properties set there will be appended to the generated node configuration.
+
+{{< note >}}
+If you add a property to the additional configuration file that has already been created by the `deployNodes` task, both properties will be present in generated node configuration.
+{{< /note >}}
+
+Alternatively, you can also add the path to the additional configuration file while running the gradle task via the `-PconfigFile` command-line option. However, this will result in the same configuration file being applied to all nodes.
+
+Following on from the previous example, the `PartyB` node in the next example below has additional configuration options added from a file called `none-b.conf`:
+
+```groovy
+task deployNodes(type: net.corda.plugins.Cordform, dependsOn: ['jar']) {
+    [...]
+    node {
+        name "O=PartyB,L=New York,C=US"
+        [...]
+        // Grants user1 the ability to start the MyFlow flow.
+        rpcUsers = [[ user: "user1", "password": "test", "permissions": ["StartFlow.net.corda.flows.MyFlow"]]]
+        configFile = "samples/trader-demo/src/main/resources/node-b.conf"
+    }
+}
+```
+
+The `drivers` Cordform parameter in the `node` entry lists paths of the files to be copied to the `drivers` sub-directory of the node.
+To copy the same file to all nodes, define `ext.drivers` in the top level, and reuse it for each node by setting `drivers=ext.drivers`.
+
+```groovy
+task deployNodes(type: net.corda.plugins.Cordform, dependsOn: ['jar']) {
+    ext.drivers = ['lib/my_common_jar.jar']
+    [...]
+    node {
+        name "O=PartyB,L=New York,C=US"
+        [...]
+        drivers = ext.drivers + ['lib/my_specific_jar.jar']
+    }
+}
+```
+
+### Package namespace ownership
+
+To configure [package namespace ownership](../../4.8/enterprise/node/deploy/env-dev.html#package-namespace-ownership), use the optional `networkParameterOverrides` and `packageOwnership` blocks, in a similar way to how the configuration file is used by the [Network Bootstrapper](network-bootstrapper.md) tool. For example:
```

Comment:
Can you treat this as a step that does not require reference to other steps?

Reply:
I'm going to flag this as a separate documentation ticket because it's going to require a convo with an SME who is as-yet unidentified.